src/search/redis_query_transformer.h

```diff
@@ -53,7 +53,32 @@ StatusOr<std::unique_ptr<parse_tree::node>> ParseToTree(Input&& in) {
 struct Transformer : ir::TreeTransformer {
   explicit Transformer(const ParamMap& param_map) : TreeTransformer(param_map) {}
 
+  StatusOr<std::unique_ptr<VectorLiteral>> Transform2Vector(const TreeNode& node) {
+    std::string vector_str = GET_OR_RET(GetParam(node));
+
+    auto vector_chars = GET_OR_RET(Binary2Chars(vector_str));
+    std::vector<double> values = GET_OR_RET(CharsToVector<double>(vector_chars));
```

Comment:
I think we can just create a function to convert the binary data to `std::vector<double>` since it's more intuitive.

Also in the SemaChecker we need to check if the length of the vector is valid.